CONTINUOUS_INTEGRATION.md

```diff
@@ -0,0 +1,198 @@
+
+```
+~/DeepSpeech/ci_scripts$ ls
+total 164
+ 4 drwxrwxr-x  2 root root  4096 Apr 20 22:52 ./
+ 4 drwxrwxr-x 16 root root  4096 Apr 20 22:52 ../
+ 4 -rwxrwxr-x  1 root root  3678 Apr 20 22:52 all-utils.sh
+ 4 -rwxrwxr-x  1 root root  2472 Apr 20 22:52 all-vars.sh
+24 -rwxrwxr-x  1 root root 23668 Apr 20 22:52 asserts.sh
+ 4 -rwxrwxr-x  1 root root  1300 Apr 20 22:52 build-utils.sh
+ 4 -rwxrwxr-x  1 root root   382 Apr 20 22:52 cpp-bytes-tests.sh
+ 4 -rwxrwxr-x  1 root root   488 Apr 20 22:52 cpp-tests-prod.sh
+ 4 -rwxrwxr-x  1 root root   345 Apr 20 22:52 cpp-tests.sh
+ 4 -rwxrwxr-x  1 root root   428 Apr 20 22:52 cpp_tflite_basic-tests.sh
+ 4 -rwxrwxr-x  1 root root   569 Apr 20 22:52 cpp_tflite-tests-prod.sh
+ 4 -rwxrwxr-x  1 root root   541 Apr 20 22:52 cpp_tflite-tests.sh
+ 4 -rwxrwxr-x  1 root root   318 Apr 20 22:52 cppwin-tests.sh
+ 4 -rwxrwxr-x  1 root root   467 Apr 20 22:52 cppwin_tflite-tests.sh
+ 4 -rw-rw-r--  1 root root   316 Apr 20 22:52 docs-requirements.txt
+ 4 -rwxrwxr-x  1 root root   602 Apr 20 22:52 electronjs-tests-prod.sh
+ 4 -rwxrwxr-x  1 root root   384 Apr 20 22:52 electronjs-tests.sh
+ 4 -rwxrwxr-x  1 root root   631 Apr 20 22:52 electronjs_tflite-tests-prod.sh
+ 4 -rwxrwxr-x  1 root root   523 Apr 20 22:52 electronjs_tflite-tests.sh
+ 4 -rwxrwxr-x  1 root root   538 Apr 20 22:52 host-build.sh
+ 4 -rwxrwxr-x  1 root root   556 Apr 20 22:52 node-tests-prod.sh
+ 4 -rwxrwxr-x  1 root root   343 Apr 20 22:52 node-tests.sh
+ 4 -rwxrwxr-x  1 root root   591 Apr 20 22:52 node_tflite-tests-prod.sh
+ 4 -rwxrwxr-x  1 root root   482 Apr 20 22:52 node_tflite-tests.sh
+ 4 -rwxrwxr-x  1 root root   614 Apr 20 22:52 package.sh
+ 4 -rwxrwxr-x  1 root root  2787 Apr 20 22:52 package-utils.sh
+ 4 -rwxrwxr-x  1 root root   521 Apr 20 22:52 python-tests-prod.sh
+ 4 -rwxrwxr-x  1 root root   274 Apr 20 22:52 python-tests.sh
+ 4 -rwxrwxr-x  1 root root   502 Apr 20 22:52 python_tflite-tests-prod.sh
+ 4 -rwxrwxr-x  1 root root   413 Apr 20 22:52 python_tflite-tests.sh
+ 4 -rwxrwxr-x  1 root root  2688 Apr 20 22:52 tf-build.sh
+ 4 -rwxrwxr-x  1 root root  2384 Apr 20 22:52 tf-package.sh
+ 4 -rwxrwxr-x  1 root root  3371 Apr 20 22:52 tf-setup.sh
+12 -rwxrwxr-x  1 root root  8375 Apr 20 22:52 tf-vars.sh
+```
+
+### Runners
+
+A _Runner_ is a server that listens for available _Jobs_ to run. The _Runner_ runs each _Job_ one at a time and then reports the results back to GitHub. A _Runner_ can be installed on a cloud server, or you can use the _Runner_ hosted by GitHub.
+
+If you are building DeepSpeech on particular hardware, or a non-Ubuntu variant of Linux (such as Arch), then you will need to [host your own _Runner_](https://docs.github.com/en/actions/hosting-your-own-runners).
+
+## Running continuous integration from GitHub
+
+In this example, we're going to assume that you will use a _Runner_ hosted on GitHub to perform continuous integration.
```

Comment:
It's always going to be true :)

Reply:
Well, possibly. Some advanced users may want to run DeepSpeech internally on their own infrastructure, and then they may need to integrate with internal CI tools. That's well beyond the scope of the PlayBook, but I think it's important to flag it as an _assumption_.

Reply:
Sure but I'd assume people doing this dont really need this sentence, but its there so keep it if you think it can be useful😉